monitor/lendingPoolLevel.js

```diff
@@ -0,0 +1,287 @@
+import axios from "axios";
+
+export const lendingPoolLevel = async (deployments) => {
+    const endpointsList = [];
+    Object.keys(deployments).forEach((depo) => {
+        if (
+            !deployments[depo].indexingError &&
+            deployments[depo].protocolType.toUpperCase() === "LENDING"
+        ) {
+            endpointsList.push(deployments[depo].url);
+        }
+    });
+
+    const baseQuery = `
+    query MyQuery {
+      markets(first: 10) {
+        id
+        name
+        totalValueLockedUSD
+        cumulativeSupplySideRevenueUSD
+        cumulativeProtocolSideRevenueUSD
+        cumulativeTotalRevenueUSD
+        cumulativeDepositUSD
+        cumulativeBorrowUSD
+        cumulativeLiquidateUSD
+        totalBorrowBalanceUSD
+        totalDepositBalanceUSD
+        outputTokenSupply
+        outputTokenPriceUSD
+      }
+    }`;
+
+    const promiseArr = [];
+
+    endpointsList.forEach((endpoint) => {
+        promiseArr.push(
+            axios.post(
+                endpoint,
+                { query: baseQuery },
+                {
+                    headers: {
+                        "Content-Type": "application/json",
+                        Accept: "application/json",
+                    },
+                }
+            )
+        );
+    });
+
+    let marketLevelData = [];
+    await Promise.all(promiseArr)
+        .then(
+            (response) =>
+            (marketLevelData = response.map((marketData) => {
+                return {
+                    markets: marketData?.data?.data?.markets || [],
+                    url: marketData.config.url,
+                    deployment: marketData.config.url.split("messari/")[1],
+                };
+            }))
+        )
+        .catch((err) => console.log(err));
+
+    marketLevelData.forEach((protocol) => {
+        if (!protocol?.markets) return;
+        const data = protocol.markets;
+        if (!data?.length) return;
+        const url = protocol.url;
+        // const dataFields = Object.keys(data)
+
+        const idx = Math.floor(Math.random() * data.length);
```

Comment:
This was from the first draft of the "Lending" protocol pool level checks, just to grab a random pool and perform some tests on it. Once I start revising upon these checks it will loop through all pools so this wont be needed.